octopoes/tools/rename-origin-method.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import json
+import logging
+from typing import Any
+
+import click
+from xtdb_client import XTDBClient
+
+logger = logging.getLogger(__name__)
+
+
+@click.group(
+    context_settings={
+        "help_option_names": ["-h", "--help"],
+        "max_content_width": 120,
+        "show_default": True,
+    }
+)
+@click.option("-n", "--node", default="0", help="XTDB node")
+@click.option(
+    "-u",
+    "--url",
+    default="http://localhost:3000",
+    help="XTDB server base url",
+)
+@click.option(
+    "-t",
+    "--timeout",
+    type=int,
+    default=5000,
+    help="XTDB request timeout (in ms)",
+)
+@click.option("-v", "--verbosity", count=True, help="Increase the verbosity level")
+@click.pass_context
+def cli(ctx: click.Context, url: str, node: str, timeout: int, verbosity: int):
+    verbosities = [logging.WARN, logging.INFO, logging.DEBUG]
+    try:
+        if verbosity:
+            logging.basicConfig(level=verbosities[verbosity - 1])
+    except IndexError:
+        raise click.UsageError("Invalid verbosity level (use -v, -vv, or -vvv)")
+
+    client = XTDBClient(url, node, timeout)
+    logger.info("Instantiated XTDB client with endpoint %s for node %s", url, node)
+
+    ctx.ensure_object(dict)
+    ctx.obj["client"] = client
+
+
+QY = '{:query {:find [(pull ?var [*])] :where [[?var :type "Origin"][?var :origin_type "observation"][?var :method]]}}'
+
+
+def method_query(method: str):
+    if method:
+        return f"""{{
+:query {{
+    :find [(pull ?var [*])] :where [
+            [?var :type "Origin"]
+            [?var :origin_type "observation"]
+            [?var :method "{method}"]
+        ]
+    }}
+}}
+"""
+    else:
+        return QY
```

Comment:
The `QY` constant (I assume this stands for `QUERY`) can be elliminated by reassigning the `method` variable below. This change will simplify the code by removing an unnecessary constant and reducing the code path:

```suggestion
def method_query(method: str):
    method = f'"{method}"' if method else ""
    return f"""{{
:query {{
    :find [(pull ?var [*])] :where [
            [?var :type "Origin"]
            [?var :origin_type "observation"]
            [?var :method {method}]
        ]
    }}
}}
"""
```

Reply:
Yes, I introduced it to have both a nice copy-paste-able one-liner and a multi-line form respecting E501... I am aware that QY can be optimized out.

Reply:
Thanks for clearing that up. I don't think adding the a copy-paste-able one-liner here is appropriate for several reasons. Such one-liners are typically personal snippets that developers use for their convenience. It doesn't necessary serve the broader team and can clutter the code, since in this case it's redundant. If a one-liner is important enough to be shared, it should be documented properly (e.g. for troubleshooting or documenting the queries and transactions behind a certain action or tool). This ensures that information is organized/ accessible and separated from the code base. Having to look this one-liner up in the script/ code is unusual and inconvenient, and it doesn't add much value to the tool/ code itself.